client-app-example/public/client-app-1/index.html

```diff
@@ -8,21 +8,37 @@ <h1>Application #1</h1>
     <h2>Routing</h2>
     <h3>Current Local Route:</h3>
     <div id="path"></div>
-    <h3>Current Host Route:</h3>
-    <div id="hostPath"></div>
+    <h3>Current Host Route Calculated From Client:</h3>
+    <div id="urlFromClientPath"></div>
 
     <h3>Change Route</h3>
     <ul>
-      <!-- internal routes URLs get remapped in js by `transformLinks` -->
-      <li><a href="#/route1">This client: /route1</a></li>
-      <li><a href="#/route2">This client: /route2</a></li>
       <!-- External URLs use full links so they open nicely in new tabs -->
-      <li>
-        <a href="//localhost:3000/#/app2/route1">Other client: /route1</a>
+      <li
+        ><a href="//localhost:3000/#/app2/route1">Other client: /route1</a>
       </li>
       <li>
         <a href="//localhost:3000/#/app2/route2">Other client: /route2</a>
       </li>
+      <!-- internal routes URLs get remapped in js by `transformLinks` -->
+      <li>
+        <a class="client-transform-link" href="#/route1">This client: /route1</a>
+      </li>
+      <li>
+        <a class="client-transform-link" href="#/route2">This client: /route2</a>
+      </li>
+    </ul>
+
+    <h3>
+      Using custom elements ifc-client-link and ifc-host-link to build full urls
+    </h3>
+    <ul>
+      <li>
+        <ifc-client-link path="#/route2">Creating a full link from the client path using ifc-client-link -- to Route 2</ifc-client-link>
```

Comment:
Minor, but for this and the host example below, I'd drop the `#` from the path and just use `/route2` and `/app2`.  It's good that things still work correctly if a `#` does get thrown in there, but I think from the user's perspective it makes more sense to only provide the path, and not a `#` that may or may not be present in the final generated URL in that position.